File: SpriteKitLib/src/main/java/br/com/insanitech/spritekit/SKActionSetTexture.java
package br.com.insanitech.spritekit;

/**
 * Created by anderson on 06/01/17.
 */

class SKActionSetTexture extends SKAction {
    private SKTexture texture;

    public SKActionSetTexture(SKTexture texture) {
        this.texture = texture;
    }

    @Override
    void computeStart() {

    }

    @Override
    void computeAction(long elapsed) {
        if (getParent() instanceof SKSpriteNode) {
            ((SKSpriteNode)getParent()).setTexture(texture);
        }
    }

    @Override
    void computeFinish() {

    }

    @Override
    boolean willHandleFinish() {
        return false;
    }
}

File: SpriteKitLib/src/main/java/br/com/insanitech/spritekit/SKActionSetTexture.kt
package br.com.insanitech.spritekit

/**
 * Created by anderson on 06/01/17.
 */

internal class SKActionSetTexture(private val texture: SKTexture) : SKAction() {
    internal override fun computeStart() {

    }

    internal override fun computeAction(elapsed: Long) {
        (this.parent as? SKSpriteNode)?.texture = texture
    }

    internal override fun computeFinish() {

    }
}
